src/base.ts

```diff
@@ -69,6 +70,7 @@ export default abstract class extends Command {
 
   async recordActionMetric(recordFunc: (recorder: Recorder) => Promise<void>) {
     try {
+      this.setSource();
```

Comment:
@peter-rr `recordActionMetric()` is called for both `asyncapi_adoption.action.invoked` and `asyncapi_adoption.action.finished`, so if there is a file, there will be a timestamp.

till now I don't see any benefit for collecting the file info at `asyncapi_adoption.action.invoked` since it will be the same as `asyncapi_adoption.action.finished` everytime.

in some commands like `asyncapi new file` it is impossible to collect this metric at `asyncapi_adoption.action.invoked` since no file has been created at that stage.

Reply:
> till now I don't see any benefit for collecting the file info at `asyncapi_adoption.action.invoked` since it will be the same as `asyncapi_adoption.action.finished` everytime.

Yeah, makes sense 👍 

> in some commands like `asyncapi new file` it is impossible to collect this metric at `asyncapi_adoption.action.invoked` since no file has been created at that stage.

Agree! However, I don't understand why the metric is not collected for commands like `asyncapi validate` or `asyncapi convert`, in which the file already exists. Might be something related to the file processing time and `asyncapi_adoption.action.invoked` is sent before the existing file is processed? 🤔